Core/include/Acts/EventData/MultiComponentBoundTrackParameters.hpp

```diff
@@ -0,0 +1,240 @@
+    static_assert(std::is_same_v<BoundSymMatrix, covariance_t> ||
+                  std::is_same_v<std::optional<BoundSymMatrix>, covariance_t>);
+    if constexpr (std::is_same_v<BoundSymMatrix, covariance_t>) {
+      for (const auto& [weight, params, cov] : cmps) {
+        m_components.push_back({weight, params, cov});
+      }
+    } else {
+      m_components = cmps;
+    }
+  }
+
+  /// Construct from a parameters vector on the surface and particle charge.
+  ///
+  /// @param surface Reference surface the parameters are defined on
+  /// @param params Bound parameters vector
+  /// @param q Particle charge
+  /// @param cov Bound parameters covariance matrix
+  ///
+  /// In principle, only the charge magnitude is needed her to allow
+  /// unambigous extraction of the absolute momentum. The particle charge is
+  /// required as an input here to be consistent with the other constructors
+  /// below that that also take the charge as an input. The charge sign is
+  /// only used in debug builds to check for consistency with the q/p
+  /// parameter.
+  MultiComponentBoundTrackParameters(
+      std::shared_ptr<const Surface> surface, const BoundVector& params,
+      ActsScalar q, std::optional<BoundSymMatrix> cov = std::nullopt)
+      : m_surface(surface), m_chargeInterpreter(std::abs(q)) {
+    m_components.push_back({1., params, cov});
+  }
+
+  /// Construct from a parameters vector on the surface.
+  ///
+  /// @param surface Reference surface the parameters are defined on
+  /// @param params Bound parameters vector
+  /// @param cov Bound parameters covariance matrix
+  ///
+  /// This constructor is only available if there are no potential charge
+  /// ambiguities, i.e. the charge type is default-constructible.
+  template <typename T = charge_t,
+            std::enable_if_t<std::is_default_constructible_v<T>, int> = 0>
+  MultiComponentBoundTrackParameters(
+      std::shared_ptr<const Surface> surface, const BoundVector& params,
+      std::optional<BoundSymMatrix> cov = std::nullopt)
+      : m_surface(surface) {
+    m_components.push_back({1., params, cov});
+  }
+
+  /// Parameters are not default constructible due to the charge type.
+  MultiComponentBoundTrackParameters() = delete;
+  MultiComponentBoundTrackParameters(
+      const MultiComponentBoundTrackParameters&) = default;
+  MultiComponentBoundTrackParameters(MultiComponentBoundTrackParameters&&) =
+      default;
+  ~MultiComponentBoundTrackParameters() = default;
+  MultiComponentBoundTrackParameters& operator=(
+      const MultiComponentBoundTrackParameters&) = default;
+  MultiComponentBoundTrackParameters& operator=(
+      MultiComponentBoundTrackParameters&&) = default;
+
+  /// Access the parameters
+  const auto& components() const { return m_components; }
+
+  /// Reference surface onto which the parameters are bound.
+  const Surface& referenceSurface() const { return *m_surface; }
+
+  /// Get a SingleBoundTrackParameters object for one component
```

Comment:
This returns at the weight and the single component ?
If so, please indicate both in the comment